appserver/payara-appserver-modules/microprofile/openapi/src/main/java/fish/payara/microprofile/openapi/impl/model/ExtensibleTreeMap.java

```diff
@@ -0,0 +1,100 @@
+package fish.payara.microprofile.openapi.impl.model;
+
+import static fish.payara.microprofile.openapi.impl.model.ExtensibleImpl.extensionName;
+
+import java.io.IOException;
+import java.lang.reflect.ParameterizedType;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.TreeMap;
+import java.util.Map.Entry;
+
+import org.eclipse.microprofile.openapi.models.Extensible;
+import org.eclipse.microprofile.openapi.models.Reference;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.databind.JsonSerializer;
+import com.fasterxml.jackson.databind.SerializerProvider;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+
+@JsonSerialize(using = ExtensibleTreeMap.ExtensibleTreeMapSerializer.class)
+public abstract class ExtensibleTreeMap<V, T extends Extensible<T>> extends TreeMap<String, V>
+        implements Extensible<T> {
+
+    @JsonIgnore
+    protected Map<String, Object> extensions = new LinkedHashMap<>();
+
+    protected ExtensibleTreeMap() {
+        super();
+    }
+
+    protected ExtensibleTreeMap(Map<String, ? extends V> items) {
+        super(items);
+    }
+
+    @Override
+    public final Map<String, Object> getExtensions() {
+        return extensions;
+    }
+
+    @Override
+    public final void setExtensions(Map<String, Object> extensions) {
+        this.extensions.clear();
+        for (Entry<String, Object> entry : extensions.entrySet()) {
+            this.extensions.put(extensionName(entry.getKey()), entry.getValue());
+        }
+    }
+
+    @SuppressWarnings("unchecked")
+    @Override
+    public final T addExtension(String name, Object value) {
+        if (value != null) {
+            this.extensions.put(extensionName(name), value);
+        }
+        return (T) this;
+    }
+
+    @Override
+    public final void removeExtension(String name) {
+        this.extensions.remove(extensionName(name));
+    }
+
+    /**
+     * Custom {@link JsonSerializer} that adds both the extended {@link TreeMap} entries as well as the
+     * {@link ExtensibleTreeMap#extensions} map to the output object unless the value represents a {@link Reference} in
+     * which case only the {@link Reference#getRef()} is added.
+     */
+    static class ExtensibleTreeMapSerializer extends JsonSerializer<ExtensibleTreeMap<?,?>> {
```

Comment:
FYI: This fixes two defects in the JSON/YAML serialisation present since 1.0 where properties on `Map` extending types would be ignored as they are handled by the map serializer. Thereby extensions would not be added and in case of a reference an empty object was rendered instead of the reference.